### PR TITLE
Migrates project to Swift 4.2

### PR DESCRIPTION
**Pull request by esreli**
Migration performed following these steps:

1. Use Swift 4.2 migrator assistant.
2. Updates for use of `Bool.toggle()`
3. Updates for use of Swift 4.2 syntax unwrapping of weak self.

Other swift 4.2 migration suggestions welcomed.

### CODE REVIEW

File: data-collection/data-collection/View Controllers/Map View Controller/MaskViewController.swift [hunk 1]
```diff
@@ -40,7 +40,7 @@ class MaskViewController: UIViewController {
         let path = UIBezierPath(rect: maskView.frame)
         let maskLayer = CAShapeLayer()
         path.append(UIBezierPath(rect: view.bounds))
-        maskLayer.fillRule = kCAFillRuleEvenOdd
+        maskLayer.fillRule = CAShapeLayerFillRule.evenOdd
```
Comment (philium) [inline data-collection/data-collection/View Controllers/Map View Controller/MaskViewController.swift]: Consider using an implicit member expression here.

File: data-collection/data-collection/View Controllers/Related Records View Controllers/Related Records Popups View Controller/RelatedRecordsPopupsViewController+Delegate.swift [hunk 2]
```diff
@@ -129,7 +129,7 @@ extension RelatedRecordsPopupsViewController {
         
         // Add an edit action, if editable.
         if childPopup.isEditable {
-            let editAction = UITableViewRowAction(style: UITableViewRowActionStyle.default, title: "Edit") { [weak self] (action, indexPath) in
+            let editAction = UITableViewRowAction(style: UITableViewRowAction.Style.default, title: "Edit") { [weak self] (action, indexPath) in
```
Comment (philium) [inline data-collection/data-collection/View Controllers/Related Records View Controllers/Related Records Popups View Controller/RelatedRecordsPopupsViewController+Delegate.swift]: Consider using an implicit member expression here.

File: data-collection/data-collection/View Controllers/Related Records View Controllers/Related Records Popups View Controller/RelatedRecordsPopupsViewController+Delegate.swift [hunk 3]
```diff
@@ -139,7 +139,7 @@ extension RelatedRecordsPopupsViewController {
 
         // Add delete action, if can delete.
         if let feature = childPopup.geoElement as? AGSArcGISFeature, let featureTable = feature.featureTable as? AGSArcGISFeatureTable, featureTable.canDelete(feature) {
-            let deleteAction = UITableViewRowAction(style: UITableViewRowActionStyle.default, title: "Delete") { [weak self] (action, indexPath) in
+            let deleteAction = UITableViewRowAction(style: UITableViewRowAction.Style.default, title: "Delete") { [weak self] (action, indexPath) in
```
Comment (philium) [inline data-collection/data-collection/View Controllers/Related Records View Controllers/Related Records Popups View Controller/RelatedRecordsPopupsViewController+Delegate.swift]: Consider using an implicit member expression here.

**Comment by esreli** [comment]
@philium thanks for the feedback. Will you have another look?

**Pull request merged**